syft/rekor/query.go

```diff
@@ -0,0 +1,211 @@
+package rekor
+
+import (
+	"context"
+	"crypto/sha1" //nolint:gosec // sha1 needed for checksums in spdx format
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/source"
+	"github.com/sigstore/cosign/pkg/cosign"
+	"github.com/sigstore/rekor/pkg/generated/client"
+	"github.com/sigstore/rekor/pkg/generated/client/entries"
+	"github.com/sigstore/rekor/pkg/generated/client/index"
+	"github.com/sigstore/rekor/pkg/generated/models"
+)
+
+// getSbom attempts to retrieve the SBOM from the URI.
+//
+// Precondition: att is not nil and client is not nil
+func getSbom(att *InTotoAttestation, client *http.Client) (*[]byte, error) {
+	if len(att.Predicate.Sboms) > 1 {
+		log.Info("attestation found on Rekor with multiple SBOMS, which is not currently supported. Proceeding with the first SBOM.")
+	}
+	uri := att.Predicate.Sboms[0].URI
+	if uri == "" {
+		return nil, errors.New("uri of sbom is empty")
+	}
+
+	resp, err := client.Get(uri)
+	if err != nil {
+		return nil, fmt.Errorf("error making http request: %w", err)
+	}
+
+	if resp == nil {
+		return nil, errors.New("http response body is nil")
+	}
+	bytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("error reading http response: %w", err)
+	}
+
+	return &bytes, nil
```

Comment:
Do we want the bytes to just be returned here or do we want this function to handle the unmarshalling as well
?